controllers/vpcendpoint/validation.go

```diff
@@ -274,3 +276,48 @@ func (r *VpcEndpointReconciler) validateExternalNameService(ctx context.Context,
 
 	return nil
 }
+
+// validatePrivateHostedZone attempts to find if the AddtlHostedZoneName has a zone, and if that zone is private, creating it if not found.
+func (r *VpcEndpointReconciler) validatePrivateHostedZone(ctx context.Context, resource *avov1alpha1.VpcEndpoint) error {
+	// AddtlHostedZoneName is optional, return if not present
+	if resource.Spec.AddtlHostedZoneName == "" {
+		return nil
+	}
+	//TODO: Error handling during private zone lookup
+	zoneOut, err := r.awsClient.GetDefaultPrivateHostedZoneId(ctx, resource.Spec.AddtlHostedZoneName)
```

Comment:
At minimum this, friendlier error message always welcome :D
```suggestion
	zoneOut, err := r.awsClient.GetDefaultPrivateHostedZoneId(ctx, resource.Spec.AddtlHostedZoneName)
	if err != nil {
	  return err
	}
```